### PR TITLE
Fixes dbroemme/darrens-simple-webapp#6

### PR DESCRIPTION
**Pull request by dbroemmer**
Fix the guess count issue

### CODE REVIEW

File: src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java [hunk 1]
```diff
@@ -53,6 +55,17 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
             }
         }
 
+        final String strGuessCount = (String) request.getParameter(GUESS_COUNT_KEY);
+        if (strGuessCount != null && !done) {
+            Integer guessCount = Integer.valueOf(strGuessCount);
+            request.setAttribute(GUESS_COUNT_KEY, guessCount);
```
Comment (dbroemme) [inline src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java:61]: The use of a request parameter is insecure, but for this game, its okay.

File: src/main/webapp/GuessTheNumber.jsp [hunk 2]
```diff
@@ -27,6 +31,7 @@ ${feedback}
     <label for="user_guess">Make your guess:</label>
     <input type="text" id="user_guess" name="user_guess" /><br>
     <input type="hidden" name="random_number" value="<%= randomNumber %>" />
+    <input type="hidden" name="guess_count" value="<%= guessCount + 1 %>" />
```
Comment (dbroemme) [inline src/main/webapp/GuessTheNumber.jsp:34]: Better to keep the logic in the servlet rather than the JSP

**Review by dbroemme — CHANGES_REQUESTED** [review]
One minor comment and one blocking comment

File: src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java (hunk 1 above)
Comment (dbroemme) [reply src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java:61]: Sounds good, will leave as is

Comment (dbroemme) [reply src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java:61]: Ok, will leave as is

File: src/main/webapp/GuessTheNumber.jsp (hunk 2 above)
Comment (dbroemme) [reply src/main/webapp/GuessTheNumber.jsp:34]: Moved increment logic to servlet

**Review by dbroemme — COMMENTED** [review]
Addressed comments

File: src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java (hunk 1 above)
Comment (dbroemmer) [reply src/main/java/com/thebroemmergroup/servlet/NumberGuessServlet.java:61]: Will leave as is

**Review by dbroemmer — COMMENTED** [review]
Addressed comments

File: src/main/webapp/GuessTheNumber.jsp (hunk 2 above)
Comment (dbroemmer) [reply src/main/webapp/GuessTheNumber.jsp:34]: Moved logic to servlet

**Review by dbroemmer — COMMENTED** [review]
Addressed jsp servlet comment

**Review by dbroemme — APPROVED** [review]
Looks great!

**Pull request merged**